core/dbt/include/global_project/macros/materializations/incremental/on_schema_change.sql

```diff
@@ -57,7 +65,8 @@
   {%- set target_columns = adapter.get_columns_in_relation(target_relation) -%}
   {%- set source_not_in_target = diff_columns(source_columns, target_columns) -%}
   {%- set target_not_in_source = diff_columns(target_columns, source_columns) -%}
-  
+  {%- set in_target_and_source = intersect_columns(target_columns, source_columns) -%}
```

Comment:
We need to add in `source_columns` here, as that's what we'll want to use for `dest_columns` after `process_schema_changes` has processed